### DISCUSSION TITLE
Using tiddlyclip along with the official twitter plug in

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#66] [Q&A] [answered]
I tried to use the official twitter plug in with tiddlyclip.
For example, 

![image](https://user-images.githubusercontent.com/67494083/151596533-568ee8e2-7779-4cb6-84a1-6b6c7b7cebb3.png)

This is how twitter plug in shows tweets within TW.

I was trying to make use of this twitter plug in while i am clipping text from twitter using tiddlyclip.

So i made a few changes to the snip rule

![image](https://user-images.githubusercontent.com/67494083/151596837-85483e2a-2a6a-43c2-8b75-4d3e05306102.png)

![image](https://user-images.githubusercontent.com/67494083/151598662-f1cbaeed-1b41-4179-a46e-e2df7147ff63.png)

![image](https://user-images.githubusercontent.com/67494083/151598584-a445f7b3-b5f3-473f-83c7-4d3bfa75d8ab.png)

This was not working. I guess something is wrong in my rule. Can you help

**Comment by buggyj** [comment, marked as answer]
the rule should be

`#rule=/[\s\S]*\/(.*?)$/g`
`#term=$1`
`#tweetID=((*@pageRef/#rule/#term*))`
`$text=<$twitter type='tweet' tweetID='((*#tweetID*))'>`
`$location=((*@pageRef*))`
`$when=((*@dateTimeShort*))`

**Reply by buggyj** [reply]
![image](https://user-images.githubusercontent.com/2855795/151610537-2fb37bd6-3c8b-449d-a0f9-1607ec1c3fc1.png)

**Reply by arunnbabu81** [reply]
This is working. But i noticed one problem. Suppose i am snipping or pinning from the same twitter thread multiple times and finally when i use the above given tweetRule, the twitter widget gets added only as seperate tiddler. Instead if i use the tweetRule first and then snip or pin multiple times from the same twitter thread, the snips and pins are not seen in the view template, even though those snips/pins are seen in the edit template.

![image](https://user-images.githubusercontent.com/67494083/151622027-98d6d388-3c51-4a37-90f2-8af594301e5d.png)

![image](https://user-images.githubusercontent.com/67494083/151622063-4fcfbf85-a9e7-45b8-812b-dc035f417383.png)

**Comment by buggyj** [comment]
in the TweetSnip replace the 'inc' with 'append'

**Reply by arunnbabu81** [reply]
I replaced the inc with append

![image](https://user-images.githubusercontent.com/67494083/151676987-7c2f93c9-56df-4c3e-b698-5c800ab173fb.png)


Still the snips and pins done after using the tweetsnip rule are not displayed in the viewtemplate even though they are seen in edit template

![image](https://user-images.githubusercontent.com/67494083/151677041-72579bac-3789-4de7-a24f-513ff84800c3.png)

![image](https://user-images.githubusercontent.com/67494083/151677049-baaa9bbc-ab5b-4417-bad1-6d83fb70f866.png)

But if i manually cut the tweetsnip from the beggining and paste it to the end of the tiddler, the other snips and pins gets visualised in the view template.

![image](https://user-images.githubusercontent.com/67494083/151677095-4b4a853d-9d1a-4edb-b680-88aa6ff7302c.png)


![image](https://user-images.githubusercontent.com/67494083/151677145-d34f154b-4d89-46e9-9f26-d4af0acdcc2c.png)


![image](https://user-images.githubusercontent.com/67494083/151677164-86b99c8a-c187-46da-80ce-fe2f699ddf05.png)

**Comment by buggyj** [comment]
Try adding a new paragraph to the end of the tweet

`$text=<$twitter type='tweet' tweetID='((*#tweetID*))'>\n\n`

**Comment by buggyj** [comment]
If that doesn't work try adding a paragraph before as well,

`$text=\n\n<$twitter type='tweet' tweetID='((*#tweetID*))'>\n\n`

**Reply by arunnbabu81** [reply]
I tried both methods. But its not working.

**Comment by buggyj** [comment]
`$text=<$twitter type='tweet' tweetID='((*#tweetID*))'/>`

**Reply by arunnbabu81** [reply]
> `$text=<$twitter type='tweet' tweetID='((*#tweetID*))'/>`

This one works.
